src/App.tsx

```diff
@@ -15,36 +15,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabet = 'alphabet',
+  Length = 'length',
+  Default = '',
+}
+
+const sortGoods = (
+  goods: string[],
+  sortType: SortType,
+  isReverse: boolean,
+): string[] => {
+  let sortedGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.Alphabet:
+      sortedGoods = sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.Length:
+      sortedGoods = sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...goods];
```

Comment:
```suggestion
    case SortType.Alphabet:
      sortedGoods.sort((a, b) => a.localeCompare(b));
      break;
    case SortType.Length:
      sortedGoods.sort((a, b) => a.length - b.length);
      break;
    default:
      break;
```
`sort` mutates an array, so you don't need to assign a new value for the variable. And you don't need to create a copy one more time